webapp/src/pages/RegisterPage/RegisterPage.tsx

```diff
@@ -0,0 +1,194 @@
+import { useReducer, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import {
+  Button,
+  ButtonVariant,
+  Input,
+  InputVariant,
+  Text,
+  TextColor,
+  TextVariant,
+  TextWeight,
+} from '../../shared/components';
+import { RegisterUserDto, ResponseError } from '../../shared/generated';
+import { authApi } from '../../shared/services/ApiService';
+import { SubmitStatus } from '../../shared/types';
+import { LOGIN_OPTIONS_URL } from '../../shared/urls';
+import './RegisterPage.css';
+
+export default function RegisterPage() {
+  const initialuserRegisterUserDto: RegisterUserDto = {
+    username: '',
+    fullName: '',
+    email: '',
+    password: '',
+    confirmationPassword: '',
+  };
+  const [userRegisterUserDto, updateUserRegisterUserDto] = useReducer(
+    (userRegisterUserDto: RegisterUserDto, updateUserRegisterUserDto: any) => ({
+      ...userRegisterUserDto,
+      ...updateUserRegisterUserDto,
+    }),
+    initialuserRegisterUserDto,
+  );
+  const [status, setStatus] = useState<SubmitStatus>({
+    type: 'pending',
+    message: '',
+  });
+  const navigate = useNavigate();
+  async function register() {
+    try {
+      await authApi.authControllerRegisterLocalUser({
+        registerUserDto: userRegisterUserDto,
+      });
+      setStatus({
+        type: 'success',
+        message: 'Registration complete',
+      });
+      setTimeout(() => {
+        navigate(LOGIN_OPTIONS_URL);
+      }, 2000);
+    } catch (error) {
+      if (userRegisterUserDto.username === '') {
+        setStatus({
+          type: 'error',
+          message: 'Username can not be empty',
+        });
+      } else if (userRegisterUserDto.email === '') {
+        setStatus({
+          type: 'error',
+          message: 'Email can not be empty',
+        });
+      } else if (userRegisterUserDto.fullName === '') {
+        setStatus({
+          type: 'error',
+          message: 'Full Name can not be empty',
+        });
+      } else if (userRegisterUserDto.password === '') {
+        setStatus({
+          type: 'error',
+          message: 'Password can not be empty',
+        });
+      } else if (
+        userRegisterUserDto.password !==
+        userRegisterUserDto.confirmationPassword
+      ) {
+        setStatus({
+          type: 'error',
+          message: 'Passwords are different',
+        });
```

Comment:
suggestion: We could extract this part into a function and validate the form values on the client side before making the API call.

Something like this:
```tsx
async function register() {
  if (!hasValidFormValues()) {
    return;
  }
  try {
    ...
  } catch (error) {
    ...
  }
}
```